src/libNode/MicroBlockPreProcessing.cpp

```diff
@@ -950,6 +1063,81 @@ void Node::UpdateBalanceForPreGeneratedAccounts() {
                         << counter);
 }
 
+void Node::StartTxnProcessingThread() {
+  LOG_MARKER();
+  auto t = [this]() -> void {
+    m_mediator.m_node->m_prePrepRunning = false;
+    if (m_mediator.ToProcessTransaction()) {
+      if (m_mediator.m_ds->m_mode == DirectoryService::Mode::IDLE) {
+        m_mediator.m_node->ProcessTransactionWhenShardBackup(
+            SHARD_MICROBLOCK_GAS_LIMIT);
+        if (!AccountStore::GetInstance().SerializeDelta()) {
+          LOG_GENERAL(WARNING, "AccountStore::SerializeDelta failed");
+        }
+      } else {
+        m_mediator.m_node->ProcessTransactionWhenShardBackup(
+            m_mediator.m_ds->m_microBlockGasLimit);
+      }
+    }
+  };
+  DetachedFunction(1, t);
+}
+
+bool Node::WaitUntilTxnProcessingDone() {
+  LOG_MARKER();
+  // wait for txn processing being ready by me (backup)
+  unique_lock<mutex> lock(m_mutexCVTxnProcFinished);
+  int timeout_time = std::max(
```

Comment:
Does this also need to be updated to use `DS_TX_PROCESSING_TIMEOUT`?

Reply:
yes. good catch